instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/AwsLambdaTracer.java

```diff
@@ -69,13 +92,55 @@ private SpanBuilder createSpan(Context context, Object input, Map<String, String
   }
 
   private void setAttributes(SpanBuilder span, Context context, Object input) {
-    span.setAttribute(FAAS_EXECUTION, context.getAwsRequestId());
+    setCommonAttributes(span, context);
     if (input instanceof APIGatewayProxyRequestEvent) {
       span.setAttribute(FAAS_TRIGGER, FaasTriggerValues.HTTP.getValue());
       httpSpanAttributes.onRequest(span, (APIGatewayProxyRequestEvent) input);
     }
   }
 
+  private void setCommonAttributes(SpanBuilder span, Context context) {
+    span.setAttribute(FAAS_EXECUTION, context.getAwsRequestId());
+    String arn = getFunctionArn(context);
+    if (arn != null) {
+      span.setAttribute(FAAS_ID, arn);
+    }
+    String accountId = getAccountId(arn);
+    if (accountId != null) {
+      span.setAttribute(CLOUD_ACCOUNT_ID, accountId);
+    }
+  }
+
+  @Nullable
+  private String getFunctionArn(Context context) {
+    if (GET_FUNCTION_ARN == null) {
+      return null;
+    }
+    try {
+      return (String) GET_FUNCTION_ARN.invoke(context);
+    } catch (Throwable throwable) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private String getAccountId(@Nullable String arn) {
+    if (accountId == null) {
+      synchronized (this) {
+        if (accountId == null) {
+          accountId = new AtomicReference<>();
```

Comment:
Since we're locking and volatile don't need to use AtomicReference, it just ends up being another volatile lookup inside it.

Reply:
Hmm, that is true. Okay, I went a little overboard with the thread safety 😛